# 演習課題「例外メッセージを指定しよう」

# 右のコードでは、RPGの勇者の行動を3行表示しますが、enemies配列にアクセスする番号を０で割るため、
# 例外が発生してプログラムが強制終了してしまいます。
# このプログラムで、例外が発生した時に、標準エラー出力に下記のメッセージを表示して下さい。

# プログラムを実行して、想定どおり出力されれば演習課題クリアです！

# 期待する出力値
# 勇者は敵に遭遇した
# 勇者は勝利した

# 期待する標準エラー出力値
# その敵は表示できません

# 例外メッセージを指定しよう

begin
  enemies = ["スライム","ドラゴン","魔王"]
  number = 0

  puts "勇者は敵に遭遇した"
  puts "勇者は#{enemies[2/number]}と戦った"
rescue ZeroDivisionError => e
  STDERR.puts "その敵は表示できません"
ensure
  puts "勇者は勝利した"
end
